Frontend_Web_Development_React_Angular_Vue/React/Week 1/REACT-1.3 ES6 Features.md

```diff
@@ -0,0 +1,322 @@
+<h1 align="center">REACT-1.3 ES6 Features</h1>
+
+<br>
+<img  src="https://reactjs.org/logo-og.png"  
+ alt="react-logo" width="100%" />
+</p>
+
+# What is React? 
+
+## Introduction
+- React is a JavaScript library which is used to create single page applications. It is a declarative, efficient, and flexible JavaScript library which allows to create reusable UI components. It is developed and maintained by Facebook.
+<p>&nbsp;</p>
+
+<h1> New Featues in Javascript ES6 </h1>
+
+- ECMAScript 2015 is also known as ES6.
+
+### 1. ***let and const*** : - Both these keywords are used to declare variable. Variables declared with let can be re-assigned but cannot be redeclared within the same scope whereas those declared by const must be assigned an initial value but cannot be redeclared within the same scope.
+
+
+ ```jsx
+let opensource = 'GWOC';
+opensource = 'GirlScript';
+console.log(opensource);
+
+//output
+GirlScript
+
+
+const frontend = 'React';
+frontend = 'React with ES6';
+console.log(frontend);
+
+//Error
+Uncaught TypeError: Assignment to constant variable.
+```
+<p>&nbsp;</p>
+
+### 2. ***The Spread Operator*** : - It is denoted by ... .It is used to expand iterable objects into multiple elements and also to combine multiple arrays into one array
+ ```jsx
+const github = ["fork", "clone", "star"];
+const pr = ["pull","commit", "push"];
+console.log(...github);
+
+const combine = [...github, ...pr];
+console.log(combine);
+
+//output
+fork clone star
+['fork', 'clone', 'star', 'pull', 'commit', 'push']
+```
+
+<p>&nbsp;</p>
+
+### 3. ***Arrow functions*** : Decreases the syntax for writing functions. Now we don't require the function keyword, the return keyword, and the curly brackets
+
+ ```jsx
+// ES5
+var x = function(x, y) {
+   return x * y;
+}
+
+// ES6
+const x = (x, y) => x * y;
+```
+<p>&nbsp;</p>
+
+### 4. ***Template literals***
+ ```jsx
+ //ES5
+const opensource = {
+ name: 'GWOC',
+ year: ' 2021'
+};
+
+let msg = "I am Contributing in " + opensource.name + opensource.year;
+console.log(msg);
+
+
+//ES6
+let msg = `I am Contributing in ${opensource.name} ${opensource.year}`;
+console.log(msg);
+
+//output
+I am Contributing in GWOC 2021
+```
+<p>&nbsp;</p>
+
+### 5. ***Adding default values to the function’s parameter***
+
+```jsx
+function opensource(name = 'GWOC', year = '2021') {
+ return ` ${name} ${year}`;
+}
+
+console.log(opensource()); // GWOC 2021
+console.log(opensource('GSSoC')); // GSSoC 2021
+
+```
+<p>&nbsp;</p>
+
+### 6. ***Extracting data from an array***
+
+```jsx
+// ES5
+const evenNumbers = [2, 4, 6];
+
+const a = evenNumbers[0];
+const b = evenNumbers[1];
+const c = evenNumbers[2];
+
+console.log(a, b, c);
+//output
+2 4 6
+
+
+//ES6
+const evenNumbers = [2, 4, 6];
+
+const [x, y, z] = evenNumbers;
+
+console.log(a, b, c);
+//output
+2 4 6
+
+
+```
+<p>&nbsp;</p>
+
+### 7. ***Extracting data from a object***
```

Comment:
This can come under destructuring

Reply:
> You can also write a bit about rest operator as you have mentioned spread operator

sir i have pushed the required changes mentioned by you. Kindly review it.